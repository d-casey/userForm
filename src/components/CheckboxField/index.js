import React from 'react'

const CheckboxField = ({ name, defaultChecked, message }) => {
  return (
    <li className="checkbox">
      <input
        className="checkbox-input"
        name={name}
        type="checkbox"
        defaultChecked={defaultChecked}
      />
      <span>{message}</span>
    </li>
  )
}

export default CheckboxField
