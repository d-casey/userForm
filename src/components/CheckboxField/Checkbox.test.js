import React from 'react'
import { create } from 'react-test-renderer'
import CheckboxField from '.'

test('renders correctly when passed all details', () => {
  const register = jest.fn()
  const component = create(
    <CheckboxField
      name={'checkbox_1'}
      defaultChecked={true}
      message={'Would you like to hear from us?'}
      register={register}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
