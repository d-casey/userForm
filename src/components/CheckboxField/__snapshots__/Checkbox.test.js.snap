// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when passed all details 1`] = `
<li
  className="checkbox"
>
  <input
    className="checkbox-input"
    defaultChecked={true}
    name="checkbox_1"
    type="checkbox"
  />
  <span>
    Would you like to hear from us?
  </span>
</li>
`;
